src/codegen/compile_enhanced.ml

```diff
@@ -11955,6 +12019,22 @@ and compile_prim_invocation (env : E.t) ae p es at =
     SR.Vanilla,
     Persistence.in_upgrade env
 
+  | OtherPrim "alloc_weak_ref", [target] ->
+    SR.Vanilla,
+    compile_exp_vanilla env ae target ^^
+    E.call_import env "rts" "alloc_weak_ref"
+
+  | OtherPrim "weak_get", [weak_ref] ->
+    SR.Vanilla,
+    compile_exp_vanilla env ae weak_ref ^^
+    WeakRef.load_field env
```

Comment:
I think we have another lurking bug with weak ref. The alloc_weak_ref takes a T and stores it as T in the field without any further wrapping option injection. weakGet reads the same field value and returns an ?T. Seem wrong to me and I suspect this will crash if T is itself an option type.